receiver/splunkhecreceiver/receiver.go

```diff
@@ -231,20 +261,75 @@ func (r *splunkReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int) {
+func (r *splunkReceiver) writeSuccessResponseWithAck(ctx context.Context, resp http.ResponseWriter, eventCount int, partitionID string) {
+	if r.ackExt != nil {
```

Comment:
It's generally more common to simply return when there's nothing to do (or an error is hit), and then continue with functionality in the main body of the method, rather than putting everything in the if statement. That's my preference at least.

Reply:
That does look cleaner. Fixed